### ISSUE TITLE
Encoding error on Python 3 when using fast readers with tables with unicode

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#5966]
If I have a CSV file with:

```
a,b
café,1
```

and I try and read it with:

```
In [3]: t = Table.read('test.csv', format='ascii')

In [4]: t
Out[4]: 
<Table length=1>
 a     b  
str4 int64
---- -----
café     1
```

this works fine. However, if I specify the following I get an error:

```
In [6]: t = Table.read('test.csv', format='ascii.csv')
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
<ipython-input-6-e452f1ad871b> in <module>()
----> 1 t = Table.read('test.csv', format='ascii.csv')

/Users/tom/miniconda3/envs/dev/lib/python3.6/site-packages/astropy/table/table.py in read(cls, *args, **kwargs)
   2428         passed through to the underlying data reader (e.g. `~astropy.io.ascii.read`).
   2429         """
-> 2430         return io_registry.read(cls, *args, **kwargs)
   2431 
   2432     def write(self, *args, **kwargs):

/Users/tom/miniconda3/envs/dev/lib/python3.6/site-packages/astropy/io/registry.py in read(cls, *args, **kwargs)
    466 
    467         reader = get_reader(format, cls)
--> 468         data = reader(*args, **kwargs)
    469 
    470         if not isinstance(data, cls):

/Users/tom/miniconda3/envs/dev/lib/python3.6/site-packages/astropy/io/ascii/connect.py in io_read(format, filename, **kwargs)
     35     from .ui import read
     36     format = re.sub(r'^ascii\.', '', format)
---> 37     return read(filename, format=format, **kwargs)
     38 
     39 

/Users/tom/miniconda3/envs/dev/lib/python3.6/site-packages/astropy/io/ascii/ui.py in read(table, guess, **kwargs)
    335             fast_reader = get_reader(**fast_kwargs)
    336             try:
--> 337                 dat = fast_reader.read(table)
    338                 _read_trace.append({'kwargs': fast_kwargs,
    339                                     'status': 'Success with fast reader (no guessing)'})

/Users/tom/miniconda3/envs/dev/lib/python3.6/site-packages/astropy/io/ascii/fastbasic.py in read(self, table)
     98                                       data_start=self.data_start,
     99                                       fill_extra_cols=self.fill_extra_cols,
--> 100                                       **self.kwargs)
    101         conversion_info = self._read_header()
    102         self.check_header()

astropy/io/ascii/cparser.pyx in astropy.io.ascii.cparser.CParser.__cinit__ (astropy/io/ascii/cparser.c:4368)()

astropy/io/ascii/cparser.pyx in astropy.io.ascii.cparser.CParser.setup_tokenizer (astropy/io/ascii/cparser.c:5693)()

UnicodeEncodeError: 'ascii' codec can't encode character '\xe9' in position 7: ordinal not in range(128)
```

this is because in this case the fast readers don't deal well with unicode. But if this kind of error happens, I think we should at least fall back to the slow reader?

**Comment by bsipocz** [comment]
I've run into this very recently, too and +1 to fall back to the slow reader and give a warning.

**Comment by saimn** [comment]
Looking at encoding issues in `io.ascii` ;), here the `UnicodeEncodeError` could easily be catched to fall back to the slow reader.
To reproduce I had to specify the format explicitly, otherwise the fast reader is not used (maybe some change in the guess logic ?) : `t = Table.read('test.csv', format='ascii.csv')`

**Comment by saimn** [comment]
This looks like a duplicate of #7047, which was fixed by  #7103.

**Comment by bsipocz** [comment]
I can confirm that the example above work now, so closing this issue.
